### PR TITLE
Add filter config to mitm config

### PR DESCRIPTION
**Pull request by AndreasBurger**

### CODE REVIEW

File: model/proxy.py [hunk 1]
```diff
@@ -86,8 +86,31 @@ def _required_attributes(self):
...
+
+
+class MitmFilterConfig(ModelBase):
+    def _required_attributes(self):
+        yield from super()._required_attributes()
+        yield from [
+            'allow',
+        ]
+
+    def allowed_hosts(self):
```
Comment (ccwienk) [inline model/proxy.py]: how about "whitelisted_hosts" + add getter for "blacklisted_hosts", please

File: model/proxy.py [hunk 2]
```diff
@@ -86,8 +86,31 @@ def _required_attributes(self):
...
+        if not config:
+            return None
+        return MitmFilterConfig(config)
+
+
+class MitmFilterConfig(ModelBase):
+    def _required_attributes(self):
+        yield from super()._required_attributes()
+        yield from [
+            'allow',
```
Comment (ccwienk) [inline model/proxy.py]: whitelisted_hosts? (+blacklisted_hosts')

Comment (ccwienk) [reply model/proxy.py]: also: I think this attribute should be optional (empty set / tuple if not configured)

File: model/proxy.py [hunk 3]
```diff
@@ -86,8 +86,31 @@ def _required_attributes(self):
...
         ]
 
+    def _optional_attributes(self):
+        yield from super()._optional_attributes()
+        yield from [
+            'filter_config',
+        ]
+
     def config(self):
         return self.raw['config']
```
Comment (ccwienk) [inline model/proxy.py]: what's `config`?

File: model/proxy.py [hunk 4]
```diff
@@ -86,8 +86,31 @@ def _required_attributes(self):
             'logging',
         ]
 
+    def _optional_attributes(self):
+        yield from super()._optional_attributes()
+        yield from [
+            'filter_config',
```
Comment (ccwienk) [inline model/proxy.py]: might it be enough to inline the one (or two) attrs of filter_config?

Comment (AndreasBurger) [reply model/proxy.py]: For consistency and (admittedly a _very_ small concern) extensibility I'd rather not.

**Pull request merged**